ui/apps/platform/src/Components/CompoundSearchFilter/types.ts

```diff
@@ -507,6 +507,105 @@ export type ComplianceScanAttributeInputType = ValueOf<
+            searchTerm: 'Cluster ID',
+            inputType: 'autocomplete',
+        },
+        Name: {
+            displayName: 'Name',
+            filterChipLabel: 'Cluster Name',
+            searchTerm: 'Cluster',
+            inputType: 'autocomplete',
+        },
+    },
+} as const;
+
+export type AlertClusterSearchFilterConfig = {
+    displayName: (typeof alertClusterSearchFilterConfig)['displayName'];
+    searchCategory: (typeof alertClusterSearchFilterConfig)['searchCategory'];
+    attributes: (typeof alertClusterSearchFilterConfig)['attributes'];
+};
+
+export type AlertClusterAttribute = keyof AlertClusterSearchFilterConfig['attributes'];
+
+export type AlertClusterAttributeInputType = ValueOf<
+    AlertClusterSearchFilterConfig['attributes']
+>['inputType'];
+
+// Alert namespace search filter
+
+export const alertNamespaceSearchFilterConfig = {
+    displayName: 'Namespace',
+    searchCategory: 'ALERTS',
+    attributes: {
+        ID: {
+            displayName: 'ID',
+            filterChipLabel: 'Namespace ID',
+            searchTerm: 'Namespace ID',
+            inputType: 'autocomplete',
+        },
+        Name: {
+            displayName: 'Name',
+            filterChipLabel: 'Namespace Name',
+            searchTerm: 'Namespace',
+            inputType: 'autocomplete',
+        },
+    },
+} as const;
+
+export type AlertNamespaceSearchFilterConfig = {
+    displayName: (typeof alertNamespaceSearchFilterConfig)['displayName'];
+    searchCategory: (typeof alertNamespaceSearchFilterConfig)['searchCategory'];
+    attributes: (typeof alertNamespaceSearchFilterConfig)['attributes'];
+};
+
+export type AlertNamespaceAttribute = keyof AlertNamespaceSearchFilterConfig['attributes'];
+
+export type AlertNamespaceAttributeInputType = ValueOf<
+    AlertNamespaceSearchFilterConfig['attributes']
+>['inputType'];
+
+// Alert deployment search filter
+
+export const alertDeploymentSearchFilterConfig = {
+    displayName: 'Deployment',
+    searchCategory: 'ALERTS',
+    attributes: {
+        ID: {
+            displayName: 'ID',
+            filterChipLabel: 'Deployment ID',
+            searchTerm: 'Deployment ID',
+            inputType: 'autocomplete',
+        },
+        Name: {
+            displayName: 'Name',
+            filterChipLabel: 'Deployment Name',
+            searchTerm: 'Deployment',
+            inputType: 'autocomplete',
+        },
+    },
+} as const;
+
+export type AlertDeploymentSearchFilterConfig = {
+    displayName: (typeof alertDeploymentSearchFilterConfig)['displayName'];
+    searchCategory: (typeof alertDeploymentSearchFilterConfig)['searchCategory'];
+    attributes: (typeof alertDeploymentSearchFilterConfig)['attributes'];
+};
+
+export type AlertDeploymentAttribute = keyof AlertDeploymentSearchFilterConfig['attributes'];
+
+export type AlertDeploymentAttributeInputType = ValueOf<
+    AlertDeploymentSearchFilterConfig['attributes']
+>['inputType'];
+
```

Comment:
I'm not too fond of this approach—it seems like a lot of code to add. I think I may need to rethink the typing for the compound search filter. While it's nice to have well-typed code, if it's becoming more of a hassle than a help, it might be time to reconsider our approach.